twomes/shoppingitem/shoppingitem.go

```diff
@@ -0,0 +1,13 @@
+package shoppingitem
+
+import "github.com/energietransitie/twomes-backoffice-api/twomes/shoppingitemtype"
+
+// Interface to allow device, cloudfeed and energyquery ID in one table. we use type to rely on selecting the right table
+type ActionModel interface{}
```

Comment:
Can you elaborate more on what your aim with this is?

Reply:
We are trying to do [supertype-subtype](https://learndatamodeling.com/blog/supertype-and-subtype/) so we can FK to the `device`, `cloudfeed` or the future `energyquery` table and by setting the type to one of the tables. The empty interface here is to allow for that generalisation.

We can also create a mapping table to do this with an `ID`, `TableName` and `RowID`. Or just having 3 fields in shoppingitem to for each table. But we thought this might be the best way to be able to set all types as an item in the ShoppingList.